app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,101 @@
+class TasksController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_task, only: %i[ edit update destroy ]
+  before_action :project_manager_only, only: %i[ new create destroy ]
+
+  # GET /tasks or /tasks.json
+  def index
+    @tasks = Task.includes(:subtasks).all
+  end
+
+  # GET /tasks/1 or /tasks/1.json
+  def show
+    @task = Task.includes(:employees).includes(:subtasks).find(params[:id])
+  end
+
+  # GET /tasks/new
+  def new
+    @task = Task.new
+    gather_form_variables
+  end
+
+  # GET /tasks/1/edit
+  def edit
+    gather_form_variables
+  end
+
+  # POST /tasks or /tasks.json
+  def create
+    @task = CreateTaskService.perform(current_user, task_params)
+
+    respond_to do |format|
+      if @task.errors.any?
+        gather_form_variables
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @task.errors, status: :unprocessable_entity }
+      else
+        format.html { redirect_to task_url(@task), notice: "Task was successfully created." }
+        format.json { render :show, status: :created, location: @task }
+      end
+    end
+  end
+
+  # PATCH/PUT /tasks/1 or /tasks/1.json
+  def update
+    @task = UpdateTaskService.perform(@task, current_user, task_params)
```

Comment:
`Task` attributes `status` and (virtual) `assigned_employee_ids` require special handling; better to handle in a service and return an AR model with errors than to put BL into controller and/or model for something with so many cases.